gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -158,7 +158,12 @@ private SeekableByteChannel open(
   public void close() {
     try {
       try {
+        if (storage != null) {
+          storage.close();
+        }
         super.close();
+      } catch (Exception e) {
+        throw new RuntimeException("Error occurred while closing the storage client", e);
```

Comment:
The pattern we use at other places is to log and ignore the exception correct?

Reply:
Any reason why we are converting this to RuntimeException? Caller usually do not catch this and I think hence this does not look correct.

Reply:
I am specifically converting the exception here because as per our interface we don't throw from `close` with new client's integration they introduced resource cleanup in `close` as well as it throws too.
On second thought, we can absorb the exception here and log it too under `WARN`, as not cleaning up the resource should not cause whole workflow failure.